pom.xml

```diff
@@ -718,6 +718,10 @@
               <name>hudson.udp</name>
               <value>-1</value>
             </property>
+            <property>
+              <name>java.io.tmpdir</name>
+              <value>${project.build.directory}/tmp</value>
+            </property>
```

Comment:
Did you test on a plugin? Does Java mkdir if ${java.io.tmpdir} does not exist?

Reply:
Oh, and also, aren't we going to hit more the typical deemed-fixed-but-well-I-ve-stopped-trying-to-understand-this-issue Windows too-long-path issue? cc @jtnord

Reply:
Yup, batmat points out this is likely to blow up spectacularly for windows users....

Java is generally ok with this - but winp (or whatever that native library thing we use is called) and the funky native calls we make are generally NOT using the Unicode long path versions and are likely to go belly up.
When things like that happen it will generally cause pain and suffering as (still) not all native Windows tooling supports long paths.

Reply:
osx also placing temp in very weird place that nobody cleaning

Reply:
> Did you test on a plugin?

Yes, a couple.